common/decklist.cpp

```diff
@@ -463,32 +465,84 @@ bool DeckList::saveToFile_Native(QIODevice *device)
     return true;
 }
 
+void DeckList::removeEmptyLinesFromStart(QVector<QString> &inputs)
+{
+    // https://stackoverflow.com/questions/4642339/trimming-a-vector-of-strings
+    inputs.erase(inputs.begin(), std::find_if(inputs.begin(), inputs.end(), [](const QString& s) {return !s.isEmpty();}));
+}
+
 bool DeckList::loadFromStream_Plain(QTextStream &in)
 {
     cleanList();
+    bool inSideboard = false, titleFound = false, isSideboard;
+    int okRows = 0, blankLines = 0;
+    QVector<QString> inputs;
+
+    // Lets convert input QTextStream => QVector
+    while (!in.atEnd())
+        inputs.push_back(in.readLine().simplified().toLower());
+
+    // "Trim" the inputs of all blank lines at the start & end of paste
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Reverse vector (to remove blanks from end)
+    removeEmptyLinesFromStart(inputs);
+    std::reverse(inputs.begin(), inputs.end()); // Vector now in correct order
```

Comment:
Lines 482-489 and the `removeEmptyLinesFromStart` method looks over engineered to me.
Empty lines at start/end could just be dropped inside the first loop, and even multiple empty lines could be condensed in the same loop, and the remaining empty lines counted.
This would avoid the need of add all the lines them to the vector, reverse the vector twice, call the obscure std-iterators-with-callback-loop-whatever at line 471 and them loop the vector again at line 493, and again at line 513.
Anyway, as long as it works.. Yeah, well, that's just, like, my opinion, man (cit.)